iocore/net/SSLUtils.cc

```diff
@@ -373,7 +361,7 @@ set_context_cert(SSL *ssl)
 
   if (ctx != nullptr) {
     SSL_set_SSL_CTX(ssl, ctx);
-#if HAVE_OPENSSL_SESSION_TICKETS
+#if TS_HAVE_OPENSSL_SESSION_TICKETS
     // Reset the ticket callback if needed
     SSL_CTX_set_tlsext_ticket_key_cb(ctx, ssl_callback_session_ticket);
```

Comment:
This is kind of off-topic, but I'm wondering that why this `SSL_CTX` need to be reset session ticket callback.
It was already set when it was inserted to the map. `SSL_set_SSL_CTX()` had some bug before ?

/cc @shinrich

Reply:
I don't remember.  My comment is from 2015.  Perhaps to deal with plugin created context's but that wouldn't be complete either?  It is called in all cases even if tickets are not enabled.  I guess since we don't call the SSL_CTX_clear_options for SSL_OP_NO_TICKET nothing would happen in the no ticket enabled case.  Based on looking at the code today, I do not see a reason for the second callback call.  I'd put that on your list of cleanup items.

Reply:
I looked at this again. I agree with this is for dealing with plugin. Because `ssl_context_enable_tickets()` is directly called by `_store_ssl_ctx()`. Which means, when plugin creates `SSL_CTX` using `init_server_ssl_ctx()`,  `
SSL_CTX_set_tlsext_ticket_key_cb()` doesn't called.

I think we should move setting session ticket callback into `init_server_ssl_ctx()` like other `SSL_CTX_set_*` are called at there. 

And this idea noticed me that we can remove keyblock stuff. Because 1). we got rid of `ticket_key_name` tag from ssl_multicert.config 2 years ago (https://github.com/apache/trafficserver/commit/ec7be2916892492fa5ebfdec2de2bae9b2bf428b) and 2). `default_global_keyblock` is always used.